build/contracts/generate-oscontracts.js

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2022 balena.io
+ *
+ * Licensed under the Apache License, Vrsion 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+const _ = require('lodash')
+const fs = require('fs-extra')
+const path = require('path')
+const contrato = require('@balena/contrato')
+const yaml = require('js-yaml')
+const requireAll = require('require-all')
+
+const BUILD_DIR = path.join(__dirname, '../../../build')
+const DEST_DIR = path.join(BUILD_DIR, 'contracts')
+const BLUEPRINT_PATHS = {
+  'os-contracts': path.join(__dirname, 'blueprints/os-contracts.yaml'),
+}
+const CONTRACTS_PATH = path.join(BUILD_DIR, '../contracts/contracts')
+
+// Load arguments
+const args = process.argv.slice(2)
+if (args.length === 0) {
+  console.error(`Usage: ${process.argv[0]} ${process.argv[1]} <os version>`)
+  process.exit(1)
+}
+
+if (/^v?[0-9]+\.[0-9]+\.[0-9]+(\+rev[0-9]+)?$/.test(args[0]) == false) {
```

Comment:
NIT - can we make this regex a const at the top of the file?

Reply:
Sure